ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -42,481 +49,358 @@
+          + deleteCount + ", Duplicate Puts: " + duplicatePuts + ", Put After Delete: " + putAfterDeletes
+          + ", Duplicate Delete: " + duplicateDeletes;
+    }
+  }
+
+  public enum BlobState {
+    Valid, Deleted, Expired;
+  }
+
+  /**
+   * The different operations supported by the tool.
+   */
+  private enum Operation {
+    DumpIndex, DumpIndexSegment, VerifyIndex
+  }
+
+  /**
+   * Config for the DumpIndexTool.
+   */
+  private static class DumpIndexToolConfig {
+
+    /**
+     * The type of operation.
+     * Operations are: DumpIndex, DumpIndexSegment, VerifyIndex
+     */
+    @Config("type.of.operation")
+    final Operation typeOfOperation;
+
+    /**
+     * The path where the input is. This may differ for different operations. For e.g., this is a file for
+     * {@link Operation#DumpIndexSegment} but is a directory for {@link Operation#DumpIndex} and
+     * {@link Operation#VerifyIndex}.
+     */
+    @Config("path.of.input")
+    final File pathOfInput;
+
+    /**
+     * The path to the hardware layout file. Needed if using
+     * {@link com.github.ambry.clustermap.StaticClusterAgentsFactory}.
+     */
+    @Config("hardware.layout.file.path")
+    @Default("")
+    final String hardwareLayoutFilePath;
+
+    /**
+     * The path to the partition layout file. Needed if using
+     * {@link com.github.ambry.clustermap.StaticClusterAgentsFactory}.
+     */
+    @Config("partition.layout.file.path")
+    @Default("")
+    final String partitionLayoutFilePath;
+
+    /**
+     * A comma separated list of blob IDs that the tool should operate on. Leaving this empty indicates that the tool
+     * should work on all blobs.
+     */
+    @Config("filter.set")
+    @Default("")
+    final Set<String> filterSet;
+
+    /**
+     * The number of index entries to process every second.
+     */
+    @Config("index.entries.to.process.per.sec")
+    @Default("Long.MAX_VALUE")
+    final long indexEntriesToProcessPerSec;
+
+    /**
+     * Constructs the configs associated with the tool.
+     * @param verifiableProperties the props to use to load the config.
+     */
+    DumpIndexToolConfig(VerifiableProperties verifiableProperties) {
+      typeOfOperation = Operation.valueOf(verifiableProperties.getString("type.of.operation"));
+      pathOfInput = new File(verifiableProperties.getString("path.of.input"));
+      hardwareLayoutFilePath = verifiableProperties.getString("hardware.layout.file.path", "");
+      partitionLayoutFilePath = verifiableProperties.getString("partition.layout.file.path", "");
+      String filterSetStr = verifiableProperties.getString("filter.set", "");
+      if (!filterSetStr.isEmpty()) {
+        filterSet = new HashSet<>(Arrays.asList(filterSetStr.split(",")));
+      } else {
+        filterSet = Collections.EMPTY_SET;
+      }
+      indexEntriesToProcessPerSec =
+          verifiableProperties.getLongInRange("index.entries.to.process.per.sec", Long.MAX_VALUE, 1, Long.MAX_VALUE);
```

Comment:
same as above comment. Can we have some reasonable default value.

Reply:
Similar answer